polyfills/settings/settings.js

```diff
@@ -196,27 +195,22 @@
     removeObserver: removeObserver,
     set onsettingschange(cb) {
       this._onsettingschange = cb;
-      this._onsettingschangeId = this._onsettingschangeId ||
-                                   ++_currentRequestId;
-      var commandObject = {
-        serialize: function() {
-          return {
-            id: ++_currentRequestId,
-            data: {
-              operation: 'onsettingschange'
-            },
-            processAnswer: answer => cb(answer.data)
-          };
-        }
-      };
-      navConnPromise.
-        then(navConnHelper => navConnHelper.sendObject(commandObject));
+      // Avoid to send another request because it's useless
+      if (this._onsettingschangeId) {
+        return;
+      }
+      this._onsettingschangeId = 1;
+      var self = this;
+      navConnHelper.createAndQueueRequest({
+        operation: 'onsettingschange',
+        callback: self._onsettingschange
```

Comment:
This won't work when changing the callback (the old one will be called always). If you want to not have to create a new request always you have to write a wrapper for the callback. Something like (out of this function!).
function execOnsettingsChange(ev) {
   this._onsettingschange && typeof this._onsettingschange == 'function' && this.onsettingschange(ev);
}
....
callback: execOnSettingsChange.bind(self)
}